### ISSUE TITLE
Handling of different game versions?

### ISSUE THREAD

**Issue by derplayer** [azisyus/LoGH-Translate#1]
Hey,

i have found out, that there are two versions of the game. v1.0 and v1.1.
Some DVDs are v1.0 and some v1.1
They have differences in database.mvx strings (new strings?) and the original Translation was made for v1.0 and will crash on  v1.1.
There is also a patch for the v1.0 to update it to v1.1.
https://web.archive.org/web/20090805044931/http://gineiden-game.jp/update.html

How do we handle this?

**Comment by azisyus** [comment]
I couldn't download that update, is it dead link or just me? Let's assume there is a new version called v1.1, basically i will export, translate and repack back, biggest translate file is `label.bin` which includes almost all UI and map names.  My only concern is about character_spec.bin because it doesn't use 4096 byte blocks rather than `label.bin`. But first i need to examine that update.

**Comment by derplayer** [comment]
Yes indeed...
Seems that they took the server offline. But it still worked a few months ago.
I should still have the update somewhere on HDD to reupload.

**Comment by azisyus** [comment]
How do we know our game is not v1.1? Where does it writes and is there any obvious difference? Did you find the update in your HDD?

**Comment by derplayer** [comment]
No seems i deleted it...
One obvious difference would be the GFWL design. 1.0 has the old gray one, and 1.1 has the blueish one.
The torrent on nyaa.si is a 1.1 DVD
The one on archive.org is a 1.0 DVD

**Comment by azisyus** [comment]
Looks like my game is V1.0 because it has old gray GFWL, do you still have that V1.1 torrent? There is 2 different torrent in nyaa.si, the active one is v1.0 and probably just copy of the archive.org and other torrent is basically dead.

**Comment by derplayer** [comment]
I still have the nyaa dead one but now on recheck it is also v1.0.
But i still have the 1.1 installed. I uploaded the database.mvx: https://www.mediafire.com/file/jcff6abp90n87p7
much more files are modified.
Will look if i can find it. I think it was from nicoblog but they replaced it with v1.0 because thats the only one working with the translation and didnt crashed?

**Comment by azisyus** [comment]
I examined your v1.1 and v1.0 datatable.mvx files;

There are new translations in `label.bin` like; `総司令官` also there are differences like; `が切断しました。` to `がゲームを離脱しました。` which means there might be more changes in game structure. There is no change in packing algorithm so we are lucky. Every string in `label.bin` has it's own unique id and they tried to use v1.0 translation in v1.1 as result game crashed because they deleted those strings by using an old game file. It's not something we can't handle, all i need is v1.1 version of the game.  

I have found this [torrent](https://paste.zacks.com/paste/Xiuqu9h38K14bNwuJtZl#+3+jymyNBhXApRq-BtZaGq38Gq0Z8311FyIRyQVre2c), it has same directory structure and files with nyaa.si v1.1 torrent, but still it has old GFWL it basically proves nyaa.si torrent is actually v1.0

**Comment by derplayer** [comment]
Yeah, i have also always worked with an v1.1 game at reserve engineering, this was also the reason why i started with this project: because i found it very weird that the old translation always crashed my game and wanted to fix it.

We need to find the update.zip or the correct dvd image.

**Comment by azisyus** [comment]
I don't believe we can find that update.zip unless we send mail to microvision or something like that.
You can try to recovery that file from your HDD or you can upload only different files which can be detect easily with git as result we can have our unoffical update.

**Comment by derplayer** [comment]
I found update.zip in an chinese logh torrent, but it is pretty much dead. I tried to recover but seems to be lost, so i made a diff of my game. https://www.mediafire.com/file/z7lzyum5znidxod/LOGH_v1.1_Diff.7z/file
the only modified file, that i dont have the original file anymore is the v1.1 GinLauncher that includes my own no-dvd patch.
Also: The exe metadata version is 1.0.0.2 and for the original release it is 1.0.0.1

**Comment by azisyus** [comment]
I updated my game with your diff and as result i'm having this error;

![xlive_error](https://user-images.githubusercontent.com/26818892/88832931-815cbb00-d20c-11ea-90e3-bb5712e0b18f.png)
(it says "Failed to initialize Live.")
Game still works but this error appears every time i launch the game.

I made another no-cd patch for myself, here is [mine](https://gitlab.com/azizyus/ginei-no-cd-patch). I changed only single instruction to make sure i didn't broke anything else but as result you cannot run game while actual CD is mounted but it should be enough for developing this project.


As you mentioned before there is 2 different version of GFWL, gray one is old one and blueish is new one, the point i'm missing is GFWL is not part of the game, launcher looks for xlive.dll and that means it should be installed separatedly from game but still necessary for running game also that explains why there is a file called `XLiveRedist01.02.0241.00.msi`  between installation files.
![installation_files](https://user-images.githubusercontent.com/26818892/88836016-0944c400-d211-11ea-9758-04ea6ca34868.png)


I was using Windows7x64 when i started to this project and i have already installed games which can cause to installation of latest GFWL (GTA4 uses GFWL too), then i decided to go for Win10 and that means only GFWL game will be LoGH and as result i ended up with old GFWL because i don't have LoGH update or another game which can update my GFWL.

I found this [thread ](https://steamcommunity.com/app/209100/discussions/0/1700542332333199449/) and i installed latest GFWL from 3. [url](http://download.gfwl.xboxlive.com/content/gfwl-public/redists/production/xliveredist.msi), now i have blueish GFWL in my V1.0 game.

**Comment by sethbest** [comment]
Hey this may be long dead but I was directed to this thread when I asked on the LOGH reddit if anyone had the 1.1 game patch. There seems to be confusion here about whether the patch changes anything other that the GFWL component. Yes it does. I recall it made some significant game mechanic changes. Back in the olden days I specifically did not use the translation patch because it would crash the game when used with the 1.1 version as is mentioned in this thread.

I have not found the 1.1 version either, just sharing this info. I recall the 1.1 version significantly changed how missiles worked (expanding their area of effect) for example. I suspect it was largely balance tweaks but they did improve the game.

https://www.reddit.com/r/logh/comments/1p8neei/logh_2008_pc_game_does_anyone_have_the_11_game/

**Comment by derplayer** [comment]
For future reference: I found the official update file. It was mirrored at GFWL Title update servers. Its now preserved at https://archive.org/details/logh2008_update

**Comment by sethbest** [comment]
Legendary. I dug for hours without finding it. Thank you very much for this.

**Comment by FugueeState** [comment]
Any update on this? When using the GFWL version of this game, it always prompts to update to 1.1.
I could play offline but would vastly prefer playing online. Currently it crashes and as far as I can tell there is no updated .mvx file for 1.1 right?
Can provide files if needed.
